sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -0,0 +1,258 @@
+    jsonstr = data.to_json(orient='records')
+    if update:
+      script = _DIVE_SCRIPT_TEMPLATE.format(display_id=update,
+                                            jsonstr=jsonstr)
+      display_javascript(Javascript(script))
+    else:
+      html = _DIVE_HTML_TEMPLATE.format(display_id=self._dive_display_id,
+                                        jsonstr=jsonstr,
+                                        sprite_size=sprite_size)
+      display(HTML(html))
+
+  def _display_overview(self, data, update=None):
+    gfsg = GenericFeatureStatisticsGenerator()
+    proto = gfsg.ProtoFromDataFrames(
+        [{'name': 'data', 'table': data}])
+    protostr = base64.b64encode(proto.SerializeToString()).decode('utf-8')
+    if update:
+      script = _OVERVIEW_SCRIPT_TEMPLATE.format(
+          display_id=update,
+          protostr=protostr)
+      display_javascript(Javascript(script))
+    else:
+      html = _OVERVIEW_HTML_TEMPLATE.format(
+          display_id=self._overview_display_id,
+          protostr=protostr)
+      display(HTML(html))
+
+  def _display_dataframe(self, data, update=None):
+    if update:
+      table_id = 'table_{}'.format(update)
+      html = _DATAFRAME_PAGINATION_TEMPLATE.format(
+          dataframe_html=data.to_html(notebook=True,
+                                      table_id=table_id),
+          table_id=table_id)
+      update_display(HTML(html), display_id=update)
+    else:
+      table_id = 'table_{}'.format(self._df_display_id)
+      html = _DATAFRAME_PAGINATION_TEMPLATE.format(
+          dataframe_html=data.to_html(notebook=True,
+                                      table_id=table_id),
+          table_id=table_id)
+      display(HTML(html), display_id=self._df_display_id)
+
+  def _to_element_list(self):
+    pcoll_list = []
+    if ie.current_env().cache_manager().exists('full', self._cache_key):
+      pcoll_list, _ = ie.current_env().cache_manager().read('full',
+                                                            self._cache_key)
+    return pcoll_list
+
+  def _to_dataframe(self):
+    normalized_list = []
+    # Column name for _one_dimension_types if presents.
+    normalized_column = str(self._pcoll)
+    # Normalization needs to be done for each element because they might be of
+    # different types. The check is only done on the root level, pandas json
+    # normalization I/O would take care of the nested levels.
+    for el in self._to_element_list():
+      if self._is_one_dimension_type(el):
+        # Makes such data structured.
+        normalized_list.append({normalized_column: el})
+      else:
+        normalized_list.append(_pv_jsons_load(_pv_jsons_dump(el)))
+    # Creates a dataframe that str() 1-d iterable elements after
+    # normalization so that facets_overview can treat such data as categorical.
+    return json_normalize(normalized_list).applymap(
+        lambda x: str(x) if type(x) in (list, tuple) else x)
```

Comment:
I think this is a super useful method for InteractiveRunner users in general. Would it be possible to move this to the InteractiveRunner class so that anyone can take advantage of converting a PCollection to a Dataframe?

Reply:
We've discussed whether we are planning to expose such API to users. Temporarily, the schema support is not done and future Dataframe API work is planned for next Q1. We'll consolidate all the related work and expose such API when everything is clear.
Here the implementation is a naive python solution. So probably don't expose it for general usage yet.